downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

```diff
@@ -40,7 +40,132 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](
+
+```java
+public static Object parse(InputStream in, Type targetType) throws BError {
+};
+```
+
+##### Java APIs to provide information about runtime artifacts
+
+New runtime Java APIs are added to provide information about the active runtime artifacts.
+
+```java
+public List<Artifact> getArtifacts();
+```
+
+This returns a list of artifact instances that represent the services at runtime. An artifact instance contains a name (service name), type (only `service` is supported now), and a map of details. The details enclose the following information.
+- `listeners` - a list of listener objects attached to the service
+- `attachPoint` - the attach point specified in the service declaration (for example, `basePath` in HTTP)
+- `service` - the service object
+
+```java
+public Node getNode();
+```
+
+This returns a node instance that represents the Ballerina runtime node. A node instance contains a nodeId (self-generated unique ID), and a map of details. The details enclose the following information.
+- `balVersion` - The Ballerina version
+- `balHome` - The path of Ballerina home
+- `osName` - Name of the Operating System
+- `osVersion` - Version of the Operating System
+
+The above APIs can be called from a Ballerina environment instance. Similar to this,
+
+```java 
+import io.ballerina.runtime.api.Artifact;
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.Node;
+
+Repository repository = env.getRepository();
+List<Artifact> artifacts = repository.getArtifacts();
+Node node = repository.getNode();
+```
+
+##### Java APIs to start a new runtime and invoke a Ballerina function
+Java APIs are introduced to start a new Ballerina runtime instance for a given module and perform function invocations within the module by calling the module initialization and module start methods sequentially before any other function calls. It is recommended to call the module stop method to gracefully shut down the Ballerina runtime at the end of the program.
+
+```java
+import io.ballerina.runtime.api.Runtime;
+
+Runtime balRuntime = Runtime.from(module);
+balRuntime.init();
+balRuntime.start();
+balRuntime.invokeMethodAsync(functionName, callback, args);
+balRuntime.stop();
+```
+
+### Improvements
+
+#### Support mapping of resource and remote function parameters to `BArray` parameter of a generic native method
+A new way has been introduced to support the binding of any resource or remote function to a generic native method, regardless of the function parameters. The generic native method should be defined with a `BArray` parameter, which represents all the parameters excluding path parameters (Handling path parameters in a similar manner is supported from 2201.5.0). To avoid errors due to overloaded methods, it is recommended to define parameter type constraints as well.
```

Comment:
```suggestion
#### Support mapping of resource and remote function parameters to `BArray` parameter of a generic native method

A new way has been introduced to support the binding of any resource or remote function to a generic native method, regardless of the function parameters. The generic native method should be defined with a `BArray` parameter, which represents all the parameters excluding path parameters (Handling path parameters in a similar manner is supported from 2201.5.0). To avoid errors due to overloaded methods, it is recommended to define parameter type constraints as well.
```